### ISSUE TITLE
Hard database rebuild

### ISSUE THREAD

**Issue by yurikuzn** [espocrm/espocrm#2592]
CLI command: 

```
php command.php rebuild --hard
```

It will drop unused columns, decrease exceeding column lengths, fix index names. Recommended to have database backup before running hard rebuild.

**Comment by yurikuzn** [comment]
https://github.com/espocrm/espocrm/commit/a89ac23625d3050b90a597eeb7c79f6209f1d98d

**Comment by rabiibrahimi** [comment]
Will this work on the rebuild from UI ? this is very nice feature especially when making custom fields and end up not using all of them it is good to have such option that will reset the database and get rid of deleted fields.

**Comment by yurikuzn** [comment]
Not, as from UI it's run under webserver that implies limitations like max execution time,

**Comment by rabiibrahimi** [comment]
OK got it. 
Thanks

**Comment by rabiibrahimi** [comment]
Testing this currently, noticed it doesn't drop any unused tables. am i missing something or this is intended ? the reason why is when you add custom entities that you don't use anymore and you end up deleting then they should be dropped with such command to release database from unused tables.

**Comment by lexman1958** [comment]
@yurikuzn ,

The function as intended works fantastic. Its long waited improvement.  It will solve the a major problem we had before which was table size exceeding limitation. Thank you so much. 

Like what @rabiibrahimi says, it would be great if the rebuild also drops unused tables as well but is not really that critical. Only problem we faced by manually deleting a table is that  somehow the entity definition still lingers and creates and error when loading the instance. Perhaps this was due to persisting link with another table. Anyway, that problem was long time ago. It will definitely be great if the rebuild is enhanced to include cleaning up orphaned tables. 

Great Job and most appreciated.
